lib/assert.js

```diff
@@ -300,13 +300,15 @@ function setEquiv(a, b, strict, actualVisitedObjects) {
 
   outer: for (const val1 of a) {
     if (!b.has(val1)) {
-      // The value doesn't exist in the second set by reference, so we'll go
-      // hunting for something thats deep-equal to it. Note that this is O(n^2)
-      // complexity, and will get slower if large, very similar sets / maps are
-      // nested inside. Unfortunately there's no real way around this.
-      for (const val2 of b) {
-        if (_deepEqual(val1, val2, strict, actualVisitedObjects)) {
-          continue outer;
+      if (!strict || typeof val1 === 'object') {
```

Comment:
@josephg Yes, I’d say it’s worth it… there might be cases where the difference is not so marginal, and I think it’s okay for Node core to sacrifice a tiny bit of readability for a small performance gain.